packages/core/test/integration/query-interface/rename-table.test.ts

```diff
@@ -0,0 +1,104 @@
+import { expect } from 'chai';
+import { DataTypes } from '@sequelize/core';
+import { sequelize } from '../support';
+
+const dialect = sequelize.dialect;
+const queryInterface = sequelize.queryInterface;
+
+describe('QueryInterface#renameTable', () => {
+  describe('without schema', () => {
+    beforeEach(async () => {
+      await queryInterface.createTable('my_test_table', {
+        name: DataTypes.STRING,
+      });
+    });
+
+    it('should rename table', async () => {
+      await queryInterface.renameTable('my_test_table', 'my_test_table_new');
+      const result = await queryInterface.listTables();
+      const tableNames = result.map(v => v.tableName);
+
+      expect(tableNames).to.contain('my_test_table_new');
+      expect(tableNames).to.not.contain('my_test_table');
+    });
+  });
+
+  if (sequelize.dialect.supports.schemas) {
+    describe('with schemas', () => {
+      const schema = 'my_schema';
+      beforeEach(async () => {
+        await queryInterface.createSchema(schema);
+        await queryInterface.createTable({ tableName: 'my_test_table', schema }, {
+          name: DataTypes.STRING,
+        });
+      });
+
+      it('should rename table with schema', async () => {
+        await queryInterface.renameTable({ tableName: 'my_test_table', schema }, { tableName: 'my_test_table_new', schema });
+        const result = await queryInterface.listTables({ schema: 'my_schema' });
+        const tableNames = result.map(v => v.tableName);
+
+        expect(tableNames).to.contain('my_test_table_new');
+        expect(tableNames).to.not.contain('my_test_table');
+      });
+
+      it('should throw error when moving table to another schema if changeSchema is not set', async () => {
+        const promise = queryInterface.renameTable(
+          { tableName: 'my_test_table', schema },
+          { tableName: 'my_test_table', schema: dialect.getDefaultSchema() },
+        );
+
+        if (['db2', 'ibmi'].includes(dialect.name)) {
```

Comment:
Related to my comment about rejectInvalidOptions, use the DialectSupports from the dialects index.ts here. We'd like to limit hardcoding dialect names for when we split the dialects